### PR TITLE
TSUpdate

### PR DESCRIPTION
**Pull request by theHamsta**
Sometimes people ask how they can update their parsers. `TSUpdate` is essentially a `TSInstall` with answering yes to reinstall so people can find this functionality easier.

If it's called without arguments it reinstalls all installed parsers.

### CODE REVIEW

File: doc/nvim-treesitter.txt [hunk 1]
```diff
@@ -127,6 +127,10 @@ You can use |:TSInstall| `all` to install all parsers.
 :TSInstallInfo~
 
 List informations about currently installed parsers
+							           *:TSUpdate*
+:TSUpdate {language}~
+
+Update the installed parser of {language} or of all installed languages.
```
Comment (stsewd) [inline doc/nvim-treesitter.txt:133]:
```suggestion
Update the installed parser of {language} or all parsers if {language} is omitted.
```

Comment (theHamsta) [reply doc/nvim-treesitter.txt]: :+1: changed to `or all installed parsers if {language} is omitted`

File: lua/nvim-treesitter/install.lua [hunk 2]
```diff
@@ -192,6 +206,13 @@ M.commands = {
       "-nargs=+",
       "-complete=custom,v:lua.ts_installable_parsers"
     }
+  },
+  TSUpdate = {
+    run = M.update,
+    args = {
+      "-nargs=*",
+      "-complete=custom,v:lua.ts_installable_parsers"
```
Comment (vigoux) [inline lua/nvim-treesitter/install.lua:214]: Can't we create a custom completion based on `installed_parsers` ?

Comment (theHamsta) [reply lua/nvim-treesitter/install.lua]: I added a completion based on installed parsers.

**Comment by theHamsta** [comment]
For completeness, I added `:TSUninstall`

File: doc/nvim-treesitter.txt [hunk 3]
```diff
@@ -128,6 +128,18 @@ You can use |:TSInstall| `all` to install all parsers.
...
 List informations about currently installed parsers
 
+							           *:TSUpdate*
+:TSUpdate {language}~
+
+Update the installed parser of {language} or all installed parsers
+if {language} is omitted.
+
+							        *:TSUninstall*
+:TSUninstall {language}~
```
Comment (kyazdani42) [inline doc/nvim-treesitter.txt:138]: shouldn't these be on the same line ?

Comment (stsewd) [reply doc/nvim-treesitter.txt:138]: If they are in the same line and there is a long function everything will need to be reformatted, so it's easier to maintain in this way.

**Comment by theHamsta** [comment]
Should we export iter_cmd ? It's actually quite useful also outside of nvim-treesitter

**Pull request merged**